### PR TITLE
00-Rector: Upgrade code to PHP8.0

### PR DESCRIPTION
**Pull request by mabumusa1**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

I managed to run rector config set to upgrade the code base to be `php8.0` compatible, this PR is the first in a series of PRs that upgrade each bundle on its own. 

This PR fixes all the issues I encountered in #12322, once this is merged I will start running the rules for all the other bundles in sequence  

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Code review is required only

### CODE REVIEW

File: app/bundles/CampaignBundle/Entity/Event.php [hunk 1]
```diff
@@ -390,9 +391,8 @@ public static function loadApiMetadata(ApiMetadataDriver $metadata)
 
     /**
      * @param string $prop
-     * @param mixed  $val
      */
-    private function isChanged($prop, $val)
+    private function isChanged($prop, mixed $val)
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Entity/Event.php:395]: Why not adding all the types when in it?

File: app/bundles/CampaignBundle/Entity/Event.php [hunk 2]
```diff
@@ -619,10 +619,8 @@ public function getLog()
 
     /**
      * Get log for a contact and a rotation.
-     *
-     * @return LeadEventLog|null
      */
-    public function getLogByContactAndRotation(Contact $contact, $rotation)
+    public function getLogByContactAndRotation(Contact $contact, $rotation): ?LeadEventLog
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Entity/Event.php:623]: Same here. When we are changing the method signature, why not add all types?

File: app/bundles/CampaignBundle/Entity/Event.php [hunk 3]
```diff
@@ -654,44 +650,42 @@ public function addChild(Event $children)
...
     {
         $this->children->removeElement($children);
     }
 
     /**
-     * @return ArrayCollection|Event[]
+     * @return \Doctrine\Common\Collections\Collection<int, \Mautic\CampaignBundle\Entity\Event>|array<int, \Mautic\CampaignBundle\Entity\Event>
      */
-    public function getChildren()
+    public function getChildren(): Collection|array
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Entity/Event.php:662]: The `children` property can only hold a `Collection` value. Is the return type of `array` valid?

File: app/bundles/CampaignBundle/Entity/Event.php [hunk 4]
```diff
@@ -849,10 +832,7 @@ public function getTriggerMode()
         return $this->triggerMode;
     }
 
-    /**
-     * @param mixed $triggerMode
-     */
-    public function setTriggerMode($triggerMode)
+    public function setTriggerMode(mixed $triggerMode)
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Entity/Event.php:835]:
```suggestion
    public function setTriggerMode(?string $triggerMode): void
```

File: app/bundles/CampaignBundle/Entity/Event.php [hunk 5]
```diff
@@ -813,10 +799,7 @@ public function getTriggerIntervalUnit()
         return $this->triggerIntervalUnit;
     }
 
-    /**
-     * @param mixed $triggerIntervalUnit
-     */
-    public function setTriggerIntervalUnit($triggerIntervalUnit)
+    public function setTriggerIntervalUnit(mixed $triggerIntervalUnit)
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Entity/Event.php:802]:
```suggestion
    public function setTriggerIntervalUnit(string? $triggerIntervalUnit): void
```

**Review by escopecz — CHANGES_REQUESTED** [review]
@mabumusa1 I added a few comments just to the first file. When we are changing a method signature, we should think about that very carefully as it is a BC break change. So we should do it completely. Adding one type and leaving other params or return type empty doesn't look that good.

Also, we cannot count on the types defined in the method doc blocks. They ware copied most of the time and developers do not pay attention to them when the code works. So we must carefully evaluate every type. This can add many bugs to the product. And adding `mixed` type everywhere is the same thing as having no types at all.

**Comment by mabumusa1** [comment]
Can not be implemented at the moment, we must fix `phpdoc` blocks with the correct types before

**Pull request closed without merging**